tests/rules/test_new_lines.py

```diff
@@ -58,3 +60,28 @@ def test_dos_type(self):
         self.check('\r\n', conf)
         self.check('---\ntext\n', conf, problem=(1, 4))
         self.check('---\r\ntext\r\n', conf)
+
+    def test_auto_type(self):
+        conf = ('new-line-at-end-of-file: disable\n'
+                'new-lines: {type: auto}\n')
+
+        self.check('', conf)
+        if (
+            # Linux:
+            platform.startswith('linux') or
+            # Mac OS:
+            platform == 'darwin'
+        ):
+            self.check('\n', conf)
+            self.check('---\ntext\n', conf)
+        else:
+            self.check('\n', conf, problem=(1, 1))
+            self.check('---\ntext\n', conf, problem=(1, 4))
+
+        # Windows:
+        if (platform == 'win32'):
+            self.check('\r\n', conf)
+            self.check('---\r\ntext\r\n', conf)
+        else:
+            self.check('\r\n', conf, problem=(1, 1))
+            self.check('---\r\ntext\r\n', conf, problem=(1, 4))
```

Comment:
Adding checks that rely on the platform is not reliable nor reproduction-safe, I disagree with adding those.

A much better way would be to mock `os.linesep` to 

```python
with mock.patch('os.linesep', return_value='\n'):
    # pretend you're on Linux
    self.check('\n', conf)
    self.check('---\ntext\n', conf)
    self.check('\r\n', conf, problem=(1, 1))
    self.check('---\r\ntext\r\n', conf, problem=(1, 4))

with mock.patch('os.linesep', return_value='\r\n'):
    # pretend you're on Windows
    self.check('\n', conf, problem=(1, 1))
    self.check('---\ntext\n', conf, problem=(1, 4))
    self.check('\r\n', conf)
    self.check('---\r\ntext\r\n', conf)
```

You may need to create two distinct test functions, for the mock to reset the `os.linesep` value in-between.

Reply:
Took me way too long to fugre this out: 

we can not pass the new-line charater to `return_value`, as this makes `os.linesep` a callable and we need it to be a varaibel. This works however and I'm implementing it:

```python
with mock.patch('os.linesep', '\n'):
```

Reply:
as `new_lines.py` uses `from os import linesep` we acttually need to override that copy of `linesep` by using:

```python
with mock.patch("yamllint.rules.new_lines.linesep", '\n'):
```

alternativly we could use `import os` in `new_lines.py` (and òs.linesep` everywhere in the file)

What would you prefere? I would personally make the test more complex and keep the import small for the package code.

Reply:
```python
with mock.patch("yamllint.rules.new_lines.linesep", '\n'):
```
sounds good :+1: